### PR TITLE
mdoc: Resolves an issue with generic array extension methods.

### LINKED ISSUE #6: [mdoc] Error updating a generic extension method
If you've got an assembly with an extension method that looks like this

```csharp
    public static class ArrayX10 {
        public static bool IsAligned<T> (this T[] vect, int index) where T : struct 
        {
            return false;
        }
    }
```
You will receive the following error with `mdoc update`
```bash
mdoc: System.ArgumentNullException: Value cannot be null.
Parameter name: member
  at Mono.Documentation.MemberFormatter.GetDeclaration (Mono.Cecil.IMemberDefinition member) [0x00006] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MemberFormatter.GetDeclaration (Mono.Cecil.MemberReference mreference) [0x00007] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.AddTargets (System.Xml.XmlNode member, Mono.Documentation.DocsNodeInfo info) [0x00055] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.UpdateExtensionMethods (System.Xml.XmlElement e, Mono.Documentation.DocsNodeInfo info) [0x000d8] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.UpdateMember (Mono.Documentation.DocsNodeInfo info) [0x0024b] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.MakeMember (System.Xml.XmlDocument doc, Mono.Documentation.DocsNodeInfo info) [0x000c1] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.DoUpdateType2 (System.String message, System.Xml.XmlDocument basefile, Mono.Cecil.TypeDefinition type, System.String output, System.Boolean insertSince) [0x00376] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.StubType (Mono.Cecil.TypeDefinition type, System.String output) [0x00030] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.DoUpdateType (Mono.Cecil.TypeDefinition type, System.String basepath, System.String dest) [0x001d9] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.DoUpdateAssembly (Mono.Cecil.AssemblyDefinition assembly, System.Xml.XmlElement index_types, System.String source, System.String dest, System.Collections.Generic.HashSet`1[T] goodfiles) [0x0005d] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.DoUpdateAssemblies (System.String source, System.String dest) [0x00125] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDocUpdater.Run (System.Collections.Generic.IEnumerable`1[T] args) [0x00404] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDoc.Run (System.String[] args) [0x001f6] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
  at Mono.Documentation.MDoc.Main (System.String[] args) [0x00006] in <bdba0d7bb95b4ee498bb485f1da6983a>:0 
See `mdoc help' for more information or use --debug to diagnose.
```

### PR DESCRIPTION
**Pull request by joelmartinez**
Closes #6

### CODE REVIEW

**Comment by jonpryor** [comment]
As part of the tests I'd like to see the HTML output, to ensure that the array extension method is shown on System.Array docs.

Actually, *that* might be a sign that the declaration should be `T:System.Array` and not `T:System.Object[]`...

Rephrased: what's the point of extension methods in documentation? It's to *show* those extension methods in an "appropriate" location. *What is the appropriate location* for showing docs on `T[]`?

There isn't one, not really.

Which suggests that the closest appropriate spot would be when viewing the docs for `System.Array`.

**Comment by joelmartinez** [comment]
@jonpryor ok I've switched it to use `System.Array`. However, I'm not really sure how to test it in the html ... if you're just working with a small library, `System.Array` won't be in the xml docs for monodoc to render.

**Comment by jonpryor** [comment]
> if you're just working with a small library, System.Array won't be in the xml docs for monodoc to render.

While true, that has nothing to do with testing. :-)

The test infrastructure [already declares `System.Array`](https://github.com/mono/api-doc-tools/blob/712445f/mdoc/Test/DocTest-v1.cs#L36), and thus [already generates HTML documentation for `System.Array`](https://github.com/mono/api-doc-tools/blob/712445f/mdoc/Test/html.expected/System/Array.html).

Thus, if you were to add/move the `IsAligned<T>(T[], ...)` extension method into `DocTest-v1.cs`, I would *expect* it to be displayed in the resulting HTML `System.Array` documentation.

File: mdoc/Test/html.expected/System/Array.html [hunk 1]
```diff
@@ -293,6 +293,15 @@ <h2 class="Section">Extension Methods</h2>
                 </td>
                 <td colspan="2">
                   <b>
+                    <a href="../System/Environment.html#M:System.Environment.IsAligned``1(``0[],System.Int32)">IsAligned&lt;T&gt;</a>
```
Comment (joelmartinez) [inline mdoc/Test/html.expected/System/Array.html]: @jonpryor see above ... the method correctly shows now in the list of extension methods in the Array html.

**Pull request merged**